ivy_tests/test_ivy/test_functional/test_core/test_creation.py

```diff
@@ -1,157 +1,404 @@
 """Collection of tests for creation functions."""
 
 # global
+from enum import Flag
 import numpy as np
 import pytest
 from numbers import Number
 from hypothesis import given, strategies as st
 
 # local
 import ivy
+from ivy.functional.backends.numpy.data_type import dtype
 import ivy_tests.test_ivy.helpers as helpers
 import ivy.functional.backends.numpy as ivy_np
 
 
-# array
+# native_array - DONE
 @given(
-    dtype_and_x=helpers.dtype_and_values(ivy_np.valid_dtypes),
-    from_numpy=st.booleans(),
+    dtype_and_x=helpers.dtype_and_values(
+        ivy_np.valid_numeric_dtypes,
+        n_arrays=1,
+        min_value=None,
+        max_value=None,
+        allow_inf=False,
+        exclude_min=False,
+        exclude_max=False,
+        min_num_dims=1,
+        max_num_dims=5,
+        min_dim_size=1,
+        max_dim_size=10,
+        shape=None,
+        shared_dtype=True,),
+    as_variable=st.booleans(),
+    num_positional_args=helpers.num_positional_args(fn_name="native_array"),
+    native_array=st.booleans(),
+    container=st.booleans(),
+    instance_method=st.booleans()
 )
-def test_array(dtype_and_x, from_numpy, device, call, fw):
-    dtype, object_in = dtype_and_x
-    # to numpy
-    if from_numpy:
-        object_in = np.array(object_in)
-    # smoke test
-    ret = ivy.array(object_in, dtype=dtype, device=device)
-    # type test
-    assert ivy.is_ivy_array(ret)
-    # cardinality test
-    assert ret.shape == np.array(object_in).shape
-    # value test
-    helpers.assert_all_close(ivy.to_numpy(ret), np.array(object_in).astype(dtype))
-    # compilation test
-    if call in [helpers.torch_call]:
-        # pytorch scripting does not support string devices
-        return
+def test_native_array(
+    dtype_and_x,
+    as_variable,
+    num_positional_args,
+    native_array,
+    container,
+    instance_method,
+    fw,
+    device
+):
+    dtype, x=dtype_and_x[0], dtype_and_x[1]
```

Comment:
no need to index here, you can just write `dtype, x = dtype_and_x` and it will unpack automatically, but you should use spaces between `=`